server/src/main/java/org/apache/cassandra/sidecar/config/yaml/CacheConfigurationImpl.java

```diff
@@ -41,34 +46,50 @@ public class CacheConfigurationImpl implements CacheConfiguration
     @JsonProperty("warmup_retries")
     protected final int warmupRetries;
 
-    @JsonProperty("warmup_retry_interval_millis")
-    protected final long warmupRetryIntervalMillis;
+    protected MillisecondBoundConfiguration warmupRetryInterval;
 
     public CacheConfigurationImpl()
     {
-        this(TimeUnit.HOURS.toMillis(1), 100, true, 5, 1000);
+        this(MillisecondBoundConfiguration.parse("1h"), 100, true, 5, MillisecondBoundConfiguration.parse("60s"));
     }
 
     @VisibleForTesting
-    public CacheConfigurationImpl(long expireAfterAccessMillis, long maximumSize)
+    public CacheConfigurationImpl(MillisecondBoundConfiguration expireAfterAccess, long maximumSize)
     {
-        this(expireAfterAccessMillis, maximumSize, true, 5, 1000);
+        this(expireAfterAccess, maximumSize, true, 5, MillisecondBoundConfiguration.parse("60s"));
     }
 
-    public CacheConfigurationImpl(long expireAfterAccessMillis, long maximumSize, boolean enabled, int warmupRetries, long warmupRetryIntervalMillis)
+    public CacheConfigurationImpl(MillisecondBoundConfiguration expireAfterAccess,
+                                  long maximumSize,
+                                  boolean enabled,
+                                  int warmupRetries,
+                                  MillisecondBoundConfiguration warmupRetryInterval)
     {
-        this.expireAfterAccessMillis = expireAfterAccessMillis;
+        this.expireAfterAccess = expireAfterAccess;
         this.maximumSize = maximumSize;
         this.enabled = enabled;
         this.warmupRetries = warmupRetries;
-        this.warmupRetryIntervalMillis = warmupRetryIntervalMillis;
+        this.warmupRetryInterval = warmupRetryInterval;
     }
 
     @Override
+    @JsonProperty("expire_after_access")
+    public MillisecondBoundConfiguration expireAfterAccess()
+    {
+        return expireAfterAccess;
+    }
+
+    @JsonProperty("expire_after_access")
+    public void setExpireAfterAccess(MillisecondBoundConfiguration expireAfterAccess)
+    {
+        this.expireAfterAccess = expireAfterAccess;
+    }
+
     @JsonProperty("expire_after_access_millis")
-    public long expireAfterAccessMillis()
+    public void setExpireAfterAccessMillis(long expireAfterAccessMillis)
```

Comment:
Given the fact that we haven't made a release yet, wouldn't it be ok to just remove this?

Reply:
We can't assume no one is using Sidecar in the open source, so I think to be on the safe side and play well with open source, we should do the work to be backwards compatible.